### DISCUSSION TITLE
MiniDexed-QwertY

### DISCUSSION THREAD

**Discussion by RiNoize** [probonopd/MiniDexed#632] [Show and tell]
Hello, I loved the MiniDexed project, FM sounds were popular when I was young so they are a happy part of my subconscious.
I already made two of these MiniDexed, one with a Raspberry Pi 3 A+ and this one I present to you where I use a raspberry Pi Zero 2 and a qwerty keyboard connected to an Arduino as a midi controller.
I chose this keyboard because it has enough physical space to house the entire project.
It is a prototype that I am still working on, I show you how I am arranging the control keys and the potentiometers.
With Arduino and the qwerty keyboard we physically have 100 keys to assign and trigger CCs, SysEx, midi notes, potentiometer banks, etc.
And with the combination of the Shift, Control, Win, Alt, AltGr keys, the number of keys is multiplied, like virtual keys.

![ControladorMiniDexed (1)](https://github.com/probonopd/MiniDexed/assets/70249646/076a2996-b575-4f25-b0d0-4cb4b96b98b3)

The stickers are so that you can visually see how I am arranging, organizing the programming.

And here I explain a little what is not seen:
To select working midi channel: Shift + F1 to F8 (F1 = midi Channel 1 to F8 = midi Channel 8).
Esc, F1 to F2 select banks of potentiometers, each bank 8 potentiometers (Pots)
Potentiometers handle midi CCs and SysEx
Esc = default Bank

If you want to do this project and collaborate to make it better, let me know so I can put it on GitHub.

Thank you to everyone who makes MiniDexed possible and accessible to everyone, really thank you.

![20240329_231959](https://github.com/probonopd/MiniDexed/assets/70249646/856088bb-5aa0-4b80-9bf5-28936bdfa353)

If you want to organize and see what other ways you can distribute the controls and parameters on the qwerty keyboard, here is a PDF file and the original SVG where I edit the labels with Inkscape.

[ControladorMiniDexed.pdf](https://github.com/probonopd/MiniDexed/files/14812814/ControladorMiniDexed.pdf)
[ControladorMiniDexed.zip](https://github.com/probonopd/MiniDexed/files/14812825/ControladorMiniDexed.zip)

**Comment by diyelectromusic** [comment]
Haha!  That's great :)

So the Arduino is scanning the keyboard matrix and sending MIDI to MiniDexed.  That's a little like the approach I took with my "pseudo TX816" but I used a Raspberry Pi Pico as the interface.

I'd like to get more of MiniDexed controls exposed over SysEx to make this kind of thing easier, but haven't really decide on the best way to do it yet :)

Great stuff.

Kevin

**Reply by Kirtai** [reply]
That's *amazingly* cool.
